gce/burst-training/README.md

```diff
@@ -0,0 +1,327 @@
+dataset or if you want to perform a large-scale hyperparameter search.
+
+This guide presents a framework for executing this process on Google Cloud
+Platform. We will also see a concrete example of this framework in action. Once
+you are done with this guide, you should be able to modify the scripts that
+accompany it to accomodate your custom training jobs.
+
+If you run into any problems, please create an issue on this repository and
+[I](https://github.com/nkashy1) will gladly help you resolve them. If you would
+like to improve this guide, do not hesitate to make a pull request.
+
+- - -
+
+## Cloud infrastructure
+
+[Google Compute Engine](https://cloud.google.com/compute/) allows us to
+provision and interact with virtual machines on Google Cloud Platform, and it
+forms the basis of our burst training framework.
+
+We will use [Google Cloud Storage](https://cloud.google.com/storage/) as the
+common medium of storage, accessible both from our local environment as well as
+from our Compute Engine virtual machines.
+
+
+## Task
+
+To demonstrate the burst training workflow, we will build an *xgboost* model to
+address the [Census income
+problem](https://archive.ics.uci.edu/ml/datasets/census+income). The problem
+involves predicting a person's income level from their other Census information.
+
+We will use *xgboost*'s *scikit-learn*-like API so that we can also perform a
+hyperparameter search using
+[RandomizedSearchCV](http://scikit-learn.org/stable/modules/generated/sklearn.model_selection.RandomizedSearchCV.html).
+In this case, although the dataset itself is small, the value of burst training
+is in allowing us to perform a much more comprehensive hyperparameter search
+compared to what would be possible on even a powerful laptop.
+
+
+## Code
+
+There are three types of code in this directory:
+
+1. Code facilitating the reproduction of your development environment on a
+   Compute Engine virtual machine - [build-image.sh](./build-image.sh) and
+   [image.sh](./image.sh).
+
+2. Code which defines your training job -- this is the kind of code that you
+   write every day using frameworks like [pandas](https://pandas.pydata.org/),
+   [scikit-learn](http://scikit-learn.org/stable/), and
+   [xgboost](https://github.com/dmlc/xgboost) -
+   [census-analysis.py](./census-analysis.py)
+
+3. Code which executes your training job in a Compute Engine instance running
+   your development environment as specified in step 1 - [train.sh](./train.sh)
+   and [census-startup.sh](./census-startup.sh)
+
+The sections below describe the function of each component.
+
+
+### Reproducing your environment on Compute Engine
+
+The very first step in the burst training process is making sure that Compute
+Engine is set up to execute the code that you either already have or will be
+writing locally. In the case of our Census example, we will be using Python 2.7
+with the packages listed in [requirements.txt](./requirements.txt).
```

Comment:
This had me confused thinking that requirements.txt needed to be uploaded to the VM to be installed. 

Possibly clarify with to note that requirments.txt is only used locally: 

The very first step in the burst training process is making sure that Compute Engine is set up with the same packages as your local environment, defined in [requirements.txt](./requirements.txt) Compute Engine will execute the code that you either already have or will be writing locally. In the case of our Census example, we will be using Python 2.7 with the packages listed in [requirements.txt](./requirements.txt).

Or

[Note: requirements.txt is only used locally.]

Reply:
Done